### PR TITLE
feat(pytest): Migrate legacy `sgid` & `suid` unit tests to `PyTest`:

### LINKED ISSUE #815: Pytest: Migrate feature test `sgid`
**What would you like to be added**:
The following test should be migrated to a pytest unit:
* [features/base/test/sgid](https://github.com/gardenlinux/gardenlinux/blob/main/features/base/test/sgid)

Thereby, there are already some pytest units ([here](https://github.com/gardenlinux/gardenlinux/tree/main/tests/helper/tests)), which can be used as blueprints for the migrated test.

**Why is this needed**:
We want to migrate each feature test to a corresponding pytest. Once we have done this, we can make the `chroot` and `kvm` platform tests the new default tests for Garden Linux builds.

### LINKED ISSUE #816: Pytest: Migrate feature test `suid`
**What would you like to be added**:
The following test should be migrated to a pytest unit:
* [features/base/test/suid](https://github.com/gardenlinux/gardenlinux/blob/main/features/base/test/suid)

Thereby, there are already some pytest units ([here](https://github.com/gardenlinux/gardenlinux/tree/main/tests/helper/tests)), which can be used as blueprints for the migrated test.

**Why is this needed**:
We want to migrate each feature test to a corresponding pytest. Once we have done this, we can make the `chroot` and `kvm` platform tests the new default tests for Garden Linux builds.

### PR DESCRIPTION
**Pull request by gyptazy**
feat(pytest): Migrate legacy unit tests to `PyTest`:
 * sgid
 * suid

**How to categorize this PR?**

/kind test
/area os
/os garden-linux

**Which issue(s) this PR fixes**:
Fixes #815
Fixes #816

**Release note**:

```feature user
NONE
```

**Tests**:
Failed:
```
PASSED base/test/test_sgid_suid_files.py::test_sgid_suid_files[sgid-whitelist_files0]
FAILED base/test/test_sgid_suid_files.py::test_sgid_suid_files[suid-whitelist_files1] - AssertionError: ['/usr/lib/dbus-1.0/dbus-daemon-launch-helper,0,110']
```

Passed:
```
PASSED base/test/test_sgid_suid_files.py::test_sgid_suid_files[sgid-whitelist_files0]
PASSED base/test/test_sgid_suid_files.py::test_sgid_suid_files[suid-whitelist_files1]
```

### CODE REVIEW

File: tests/helper/tests/sgid_suid_files.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import logging
+import os
+import pytest
+from helper import utils
```
Comment (chrinorse) [inline tests/helper/tests/sgid_suid_files.py:4]: remove unneeded imports

File: features/base/test/test_sgid_suid_files.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                 "/usr/bin/newgrp,0,0",
+                 "/usr/bin/su,0,0",
+                 "/usr/lib/dbus-1.0/dbus-daemon-launch-helper,0,111",
+                 "/usr/lib/dbus-1.0/dbus-daemon-launch-helper,0,110",
+                 "/usr/bin/chfn,0,0",
+                 "/usr/bin/gpasswd,0,0",
+                 "/usr/bin/sudo,0,0",
+                 "/usr/bin/passwd,0,0"
+                 ]
+        ),
```
Comment (chrinorse) [inline features/base/test/test_sgid_suid_files.py:32]: comma not needed here

**Review by chrinorse — CHANGES_REQUESTED** [review]

File: tests/helper/tests/sgid_suid_files.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return files
+
+
+def _val_whitelist_files(remote_files, whitelist_files):
+    """ Validates that remotly found files are in whitelist """
+    found_files = []
+    for file in remote_files:
+        if file not in whitelist_files:
+            found_files.append(file)
+            logger.warn(f"Found file: {file}")
```
Comment (chrinorse) [inline tests/helper/tests/sgid_suid_files.py:35]: this is working without importing logging and initializing the logger?

File: tests/helper/tests/sgid_suid_files.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import pytest
```
Comment (chrinorse) [inline tests/helper/tests/sgid_suid_files.py:1]: as far as I can see it, import pytest is not needed here

**Review by chrinorse — CHANGES_REQUESTED** [review]

File: tests/helper/tests/sgid_suid_files.py (hunk 3 above)
Comment (gyptazy) [reply tests/helper/tests/sgid_suid_files.py]: Ah, thanks. Removed the logging line. This was for testing purpose.

**Review by chrinorse — APPROVED** [review]

**Pull request merged**